homeassistant/components/panasonic_viera/config_flow.py

```diff
@@ -0,0 +1,150 @@
+"""Config flow for Panasonic Viera TV integration."""
+from functools import partial
+import logging
+from urllib.request import URLError
+
+from panasonic_viera import TV_TYPE_ENCRYPTED, RemoteControl, SOAPError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PIN, CONF_PORT
+from homeassistant.core import callback
+
+from .const import (  # pylint: disable=unused-import
+    CONF_APP_ID,
+    CONF_ENCRYPTION_KEY,
+    CONF_ON_ACTION,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DOMAIN,
+    ERROR_INVALID_PIN_CODE,
+    ERROR_NOT_CONNECTED,
+    ERROR_UNKNOWN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Panasonic Viera."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize the Panasonic Viera config flow."""
+        self._data = {
+            CONF_HOST: None,
+            CONF_NAME: None,
+            CONF_PORT: None,
+            CONF_ON_ACTION: None,
+        }
+
+        self._remote = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is not None:
+            await self.async_load_data(user_input)
+            return await self.async_create_remote()
+
+        return self._show_user_form()
+
+    async def async_step_pairing(self, user_input=None):
+        """Handle the pairing step."""
+        if user_input is not None:
+            pin = user_input[CONF_PIN]
+            try:
+                self._remote.authorize_pin_code(pincode=pin)
+            except SOAPError as err:
+                _LOGGER.error("Invalid PIN code: %s", err)
+                return self._show_user_form({"base": ERROR_INVALID_PIN_CODE})
+            except (TimeoutError, URLError, OSError) as err:
+                _LOGGER.error("Could not establish remote connection: %s", err)
+                return self._show_user_form({"base": ERROR_NOT_CONNECTED})
+            except Exception as err:  # pylint: disable=broad-except
+                _LOGGER.exception("Unknown error: %s", err)
+                return self._show_user_form({"base": ERROR_UNKNOWN})
```

Comment:
If we need to go to another step we should await and return that step, not the form.

Reply:
then how can I make the step show the form with an error? (I'm pretty sure you already explained it but I didn't get it)

Reply:
If we want to show an error in the form of the pairing step, we set the error dict with the error key and let the logic reach the end of the step. The step should end with show form. That form call should use the populated errors dict of the current step.

Reply:
I think I'll just replace all the errors with abortions

Reply:
Oh, ok. So I can't show a previous form then?

Reply:
Here's an example:
https://github.com/home-assistant/core/blob/55dfca7467044107d6e3a654a82b8e4dd04e5afe/homeassistant/components/powerwall/config_flow.py#L40-L59

Reply:
We can move to a previous step but not show the errors of the current step in the form of the previous step.

This is why it does not make sense to store the errors as an instance attribute. The errors should always be local to the current step.